d2layouts/d2dagrelayout/layout.go

```diff
@@ -260,8 +257,10 @@ func setGraphAttrs(attrs dagreGraphAttrs) string {
 }
 
 func escapeID(id string) string {
-	// an unescaped \r will become a \n in the layout result
-	return strings.ReplaceAll(id, "\r", "\\r")
+	id = strings.ReplaceAll(id, "\\n", "\\\\n")
+	// avoid an unescaped \r becoming a \n in the layout result
+	id = strings.ReplaceAll(id, "\r", "\\r")
```

Comment:
wouldn't this actually be r?

Reply:
Note that these are being written to a JS script within a  JS template literal
`"\r"` is an actual `\r` which is replaced by `"\\r"`.
When the d2 text is `a\nb`, the ID is `\"a\\nb\"`, when it is `a\rb` the ID is `a\rb` see: https://github.com/terrastruct/d2/blob/2aa9e4b363bd7757988f8e83c54446fcd8cb85d9/d2compiler/compile_test.go#L1342-L1366 

So to get an escaped newline back `"\n"` we need the `"\\\\n"` and to get the `\r` back we need `"\\r"`